looker_style_guide.md

```diff
@@ -0,0 +1,184 @@
+
+Think of views as raw ingredients that are in the vegetable/fruit aisles in a grocery store. They are ingredients that can be combined and packaged into a product that fits our customers' needs. These "packaged items" are explores. More on that below.
+
+```
+fishtown project
+├── core
+│   └── customers.view
+├── csm
+│   └── sprints.view
+├── dbt_cloud
+│   ├── cloud_accounts.view
+│   ├── cloud_ide_sessions.view
+│   └── cloud_projects.view
+├── dbt_core
+│   ├── dbt_doc_viewer_activity.view
+│   └── dbt_projects.view
+├── finance
+│   ├── quickbook_invoices.view
+│   └── subscription_transactions.view
+└── fishtown.model
+```
+
+Explores should be optimized for the business stakeholder where we're getting the best represented business unit to run our analyses off of. This means that explores should be organized based on department or category of the business unit. For example, we have several Slack views, but maybe they all join under a singular `Slack Messages` explore, which should fall under the `Community` category.
+
+Building off of our above analogy, explores are the packaged items that can combine ingredients (views) to suit a customer's needs. We can utilize various views to create different explores.
+
+
+## Views
+
+#### Structure
+* A view's name should represent the business unit (i.e. the level of granularity of the table). For example, if we're taking `fct_intercom_conversations`, then the view name should be: `intercom_conversations`.
+* The `sql_table_name` should always have the [user attribute feature](https://blog.getdbt.com/how-to-integrate-dbt-and-looker-with-user-attributes/).
+* Dimensions and measures should be organized by group label
+
+#### Dimensions & Measures
+* Dimensions and measures should be ordered as (if fields are applicable): name, label, group_label, description, hidden, type, sql, value_format_name, filter
+* Both should have a description. This is especially important when we have common dimension/measure names across views (e.g. `created_at`). Descriptions should describe the definition, use case and/or calculation.
+* There should be a dimension for every field that exists in the table that the view is built off of. If the dimension is not useful for visualizations (e.g. an `id` field created via the surrogate key) then it should be flagged as hidden.
+* We prefer `value_format_name` over `value_format`
+* Measures should reference the dimension (e.g. ${order_total} over ${TABLE}.order_total)
+
+#### Example
+
+```
+view: intercom_conversations {
+  sql_table_name:
+  -- if prod -- analytics.analytics.fct_intercom_conversations
+  -- if dev -- analytics.{{_user_attributes['dbt_schema']}}.fct_intercom_conversations ;;
+
+# =============================================== DIMENSIONS
+
+# ==================== IDs
+  dimension: conversation_id {
```

Comment:
Should we add a comment about primary keys and why we need to define them in the LookML ?

Reply:
Added!